ext/pdo/pdo_stmt.c

```diff
@@ -1759,18 +1708,16 @@ bool pdo_stmt_setup_fetch_mode(pdo_stmt_t *stmt, zend_long mode, uint32_t mode_a
 						return false;
 					}
 					if (Z_TYPE(args[1]) == IS_ARRAY && zend_hash_num_elements(Z_ARRVAL(args[1]))) {
-						constructor_args = Z_ARRVAL(args[1]);
+						if (cep->constructor == NULL) {
+							zend_argument_value_error(3, "must be empty when class provided in argument #2 ($class) does not have a constructor");
+							return false;
+						}
+						GC_TRY_ADDREF(Z_ARRVAL(args[1]));
```

Comment:
Okay without looking too deep again: check why you refcount

Reply:
The HashTable comes from userland via the call to `POD::setFetchMode()`